### ISSUE TITLE
Flag to only take mouse inputs on Titlebar?

### ISSUE THREAD

**Issue by Bonkt** [ocornut/imgui#5304]
Using sfml I render my main application to a sf::RenderTexture that I then display on ImGui::Image(). So basically my dockspace and my whole window is at all times completely covered by ImGui begin() windows.

So when i try to dispatch mouse clicks to my application window ( my ImGui::Begin() window containing ImGui::Image where i display my rendered sfml texture.) i want to check if io.WantCaptureMouse.
The problem is that i also want to be able to move my "application" window around. So setting the flag ImGuiWindowFlags_NoMouseInputs does not work. 

I read at many places that io.WantCaptureMouse is prefered over IsWindowHovered(). 
Is this still the case for my use case? Or would a new flag with a name like
 ImGuiWindowFlags_OnlyMouseInputsOnTitlebar be a useful addition to the API?

Edit: 
Basically i want to create something similiar looking to the "Canvas" demo. But instead of rendering the canvas in ImGui i want to render in SFML to a texture and then display that on a ImGui windows. 

Also im using the docking branch, if that matters.

_Originally posted by @Bonkt in https://github.com/ocornut/imgui/issues/52#issuecomment-1119655045_

**Comment by rokups** [comment]
Could you please clarify what do you mean by "my application window"? What kind of window is that?

General rule is this:
- When `io.WantCaptureMouse == true`, you pass sfml events to imgui only, you do not process events in your own application.
- When `io.WantCaptureMouse == false`, you pass sfml events to imgui **and** you process events in your own application.

**Comment by Bonkt** [comment]
It is a ImGui::Begin() window with a single ImGui::Image() in it.
I render circuitWindowTexture in sfml in real time, then display it using Image(). 

So i want io.WantCaptureMouse = false when i am hovering the ImGui::Image(). 
And io.WantCaptureMouse = true on the Titlebar. 

Thus a request for the flag ImGuiWindowFlags_OnlyMouseInputsOnTitlebar?

`    if (ImGui::Begin(windowName.c_str(), &isOpen, window_flags))
    {
        windowSize = ImGui::GetWindowSize();
        windowPos = ImGui::GetWindowPos();
        ImGui::Image(circuitWindowTexture);
    }

    ImGui::End();`

I know and understand the "general rule" for how io.WantCaptureMouse works. I'm wondering whether the constant warnings in documentations of using IsWindowHovered() should be disregarded for what im building.

**Comment by rokups** [comment]
Ah so its only imgui windows. We have no such flag. One way to achieve what you want is to use `InvisibleButton()` to cover entire body of the window. Then `InvisibleButton()` will capture clicks, you dont act on them and window is draggable only by a titlebar.

**Comment by Bonkt** [comment]
Ahh, yes. Thats exactly what i needed. I thought i was crazy stupid being the only one that needed this. Thanks.

**Comment by ocornut** [comment]
Image();
if (IsItemHovered())
    CaptureMouseFromApp(false);

**Comment by Bonkt** [comment]
Thanks, CaptureMouseFromApp(false) is what fits best, 
I missed that function when looking through demo and the header file.
